# JUMPptm #

 * [Introduction](#introduction)
 * [Input Data](#input-data)
 * [Sample Data](#sample-data)
 * [Output Data](#output-data)

---

## Introduction ##

JUMPptm aims to identify PTM events from unmatched spectra after conventional peptide analysis of the whole proteome. By default, JUMPptm assumes that the whole proteome data has been analyzed by the JUMP suite, which outputs the identification of peptides/proteins, and the high-quality (HQ) unmatched spectra with de novo tags. Such HQ spectra (input #1) are taken as JUMPptm input, and searched against a list of PTMs specified by the user (input #2) using the multi-stage database search strategy using Comet. The PTM list can be guided by the results from an open search using MSFragger. JUMPptm exports the PTM peptide identification with TMT-based quantification.


NOTE : Please use *ptm_pipeline2.py* script to run the entire pipeline for the standalone version. Similarly, *ptm_pipeline.py* script run entire pipeline for Platform LSF to schedule jobs on the computational cluster. As the list of PTMs gets larger or number of mzXML files increases, is is better to use cluster rather than standalone version. For other platform, users may edit the job submission functions.

Note that JUMPptm only supports 64-bit macOS and Linux.



[Top of page](#JUMPptm)

----

## JUMPptm Publication:
  * [Poudel et al. Proteomics XXXX](http://willbeupdated).
  * If you use JUMPptm as part of a publication, please include this reference.

---

## Input Data ##

JUMPptm requires different input for analyzing the PTMs :
 - mzXML files 
 - PTMs list (may be derived by using open searches or PTMs of interest)
 - JUMP-search results (the program utilizes tags information. The .tags file are present inside the jump search folder)
 - JUMP filter results (Following JUMP -f, it generates a table of PSMs called ID.txt file. This file is required for selection of HQ spectrum. JUMPptm performs internal filtering to get the peptides of high confidence using FDR = 0% at peptide level)

 
NOTE: To facilitate subsequent stages of PTM analysis after default peptide analysis, the latest JUMP suite (v1.13.1; https://github.com/JUMPSuite/JUMP) outputs three files after analyzing a whole proteome dataset: i) the accepted unique proteins (.fasta format); ii) HQ unmatched spectra generated by the spectrum QC module (.ms2 format); and iii) de novo amino acid tags for each MS2 spectrum (.tags format). The identified proteins are used to generate a customized database to restrict the search space, and the latter two files are input files for JUMPptm. Note that all the HQ spectra are mass-corrected by the JUMP analysis, allowing narrow mass tolerance (e.g., < 6 ppm) for subsequent PTM identification.

